src/main/java/org/folio/rest/impl/PurchaseOrdersAPI.java

```diff
@@ -59,11 +96,133 @@ public void deleteOrdersStoragePurchaseOrdersById(String id, String lang, Map<St
     PgUtil.deleteById(PURCHASE_ORDER_TABLE, id, okapiHeaders, vertxContext, DeleteOrdersStoragePurchaseOrdersByIdResponse.class, asyncResultHandler);
   }
 
-  @Override
   @Validate
-  public void putOrdersStoragePurchaseOrdersById(String id, String lang, org.folio.rest.jaxrs.model.PurchaseOrder entity,
-      Map<String, String> okapiHeaders,
-      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
-    PgUtil.put(PURCHASE_ORDER_TABLE, entity, id, okapiHeaders, vertxContext, PutOrdersStoragePurchaseOrdersByIdResponse.class, asyncResultHandler);
+  public void putOrdersStoragePurchaseOrdersById(String id, String lang, org.folio.rest.jaxrs.model.PurchaseOrder entity, Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    try {
+      vertxContext.runOnContext(v -> {
+        Tx<PurchaseOrder> tx = new Tx<>(entity);
+        Future.succeededFuture(tx)
+          .compose(this::startTx)
+          .compose(this::deleteSequence)
+          .compose(t -> updatePurchaseOrder(tx, id, entity, okapiHeaders, vertxContext))
```

Comment:
The `Tx` holds already `entity` so makes sense to remove third param from `updatePurchaseOrder`